### PR TITLE
feat: introduce dockerChildPrefix option

### LINKED ISSUE #8238: Support unique naming schema for docker containers when running binarySource=docker
**What would you like Renovate to be able to do?**

Currently Renovate is naming docker containers based on the manager e.g. `renovate_python` for a python container.
This makes runnig multiple containers on a single host very hard without something like lxd or docker in docker o.s.

In a perfect world I would be able to run multiple renovate processes on the same node without them interfering with each other.

There is one aspect that needs to be considered.
What will happen to child containers when the process crashes?
In the current setup the next renovate run will detect the containers due to the global naming and cleans them up.
This aspect should also be supported with the new solution.

**Did you already have any implementation ideas?**

To find a good solution we need to clearify a few things:

* Renovate gives the container a certain name e.g. `renovate_python` for the `renovate/python` image.
* Renovate gives the container a certain label `renovate_child`.
* Renovate removes all containers with the label `renovate_child` at the start of any run.

With that in mind there are some ideas on how to implement this:

The simplest solution would be to add a global prefix config option that renovate would us to name the containers.
This would be easy to implement and can be set to e.g. the repository name.
This would require to also use the prefix in the `renovate_child` label so containers are not removed by any Renovate run by accident.
The drawback would be that a dangling container may never be removed when the same prefix is not used anymore.
To me this seems like acceptable drawback as long as this is also clearly communicated.

A more complex solution would include some kind of clean up routine.
E.g. remove all child container with a `renovate_child` label that are older than X days.

For the time being I would recommend (given the prefix is `foo_`):

* Add a `dockerChildPrefix` option that will add the `foo_` prefix to the label `renovate_child` and by default is empty, so the resulting label would be `foo_renovate_child`. 
* Add the prefix to the dangling cleanup function, so that the function will look only for images with the label `foo_renovate_child`.
* Add the prefix to the name of the docker image, so that the `renovate/python` image will be named `foo_renovate_docker`. This would prevent naming conflicts.
* Add documentation for the prefix and make clear that danglig containers are only removed on the next Renovate run with the same prefix.

If this is acceptable I would be happy to implement this feature in Renovate.

### PR DESCRIPTION
**Pull request by Chumper**
## Changes:

Introduces a new `dockerChildPrefix` option that influences the name and the label of docker sidecar containers.
This enables people to run Renovate as multiple processes on a single node without interfering with each others spawned docker containers.

## Context:

Closes #8238
Superseds #8265

## Documentation (please check one with an [x])

- [x] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added unit tests, or
- [x] Unit tests + ran on a real repository

### CODE REVIEW

**Comment by viceice** [comment]
why a new pr?

File: docs/usage/self-hosted-configuration.md
Comment (viceice) [inline docs/usage/self-hosted-configuration.md:105]: @rarkins What about changing all manager to only pass `node` instead of `renovate/node` this would solve the naming issue. 

And we can simply join the `dockerImagePrefix`.

**Comment by Chumper** [comment]
I created a new one, because the contributing guidelines said not to force push a PR and making the changes by hand + updating the branch to the newest head was more work than to create a new PR.

**Comment by viceice** [comment]
Ok, in such cases a force push is allowed. It only forces us to review the pr fully, because we don't know what changed. 🤗

**Comment by Chumper** [comment]
understood, next time I know :)

File: docs/usage/self-hosted-configuration.md
Comment (HonkingGoose) [inline docs/usage/self-hosted-configuration.md:103]:
```suggestion
Adds a custom prefix to the default Renovate sidecar Docker containers name and label.
```

Renovate is a noun, so should be capitalized.

File: docs/usage/self-hosted-configuration.md
Comment (rarkins) [reply docs/usage/self-hosted-configuration.md:105]: Sure

Comment (rarkins) [reply docs/usage/self-hosted-configuration.md:105]: @Chumper what do you think about this?

Comment (Chumper) [reply docs/usage/self-hosted-configuration.md:105]: I am not sure if this would solve anything.

The name of the container is derived from the registry + image, in this case `renovate/node`.
Passing the only `node` to manager would solve what problem?
The naming of the container would still be `renovate_node`; if the default is `renovate_` we would still end up with `renovate_renovate_node`.

Can you give some examples?

e.g. 
image: `renovate/node`
prefix: `myprefix_`
=> 
container_name: 
container_label

Comment (viceice) [reply docs/usage/self-hosted-configuration.md:105]: e.g.
dockerImagePrefix: ghcr.io/renovatebot
image: node
prefix: myprefix_
=>
container_name: myprefix_node
container_label: myprefix_child
container_image: ghcr.io/renovatebot/node

Comment (Chumper) [reply docs/usage/self-hosted-configuration.md:105]: ok, that would change the calculation of the container name.
Before we derived it from registry + image as pointed above, now it will just be prefix + image.

If you are fine with that than I can apply those changes.

Comment (viceice) [reply docs/usage/self-hosted-configuration.md:105]: yes, that why i proposed to only pass the image name without registry and repository from the managers.

so they pass `image=node` instead of `image=renovate/node`

Comment (rarkins) [reply docs/usage/self-hosted-configuration.md:105]: This change is still pending, right?

Comment (Chumper) [reply docs/usage/self-hosted-configuration.md:105]: yes, i didn't have the time yet to make the changes. I will update the branch once I did.

File: docs/usage/self-hosted-configuration.md
Comment (viceice) [inline docs/usage/self-hosted-configuration.md:105]:
```suggestion
If this is set to `myprefix_` the final image name for `renovate/node` would be named `myprefix_node` instead of currently used `renovate_node` and be labeled `myprefix_child` instead of `renovate_child`.
```

**Review by viceice — DISMISSED** [review]
Updated the image naming as intended 🙃

File: lib/manager/mix/__snapshots__/artifacts.spec.ts.snap
Comment (rarkins) [inline lib/manager/mix/__snapshots__/artifacts.spec.ts.snap:51]: Doesn't look right

File: lib/manager/mix/artifacts.ts
Comment (rarkins) [inline lib/manager/mix/artifacts.ts:50]: Doesn't look right. Is this because it was a legacy docker run that we hadn't migrated yet?

**Review by rarkins — CHANGES_REQUESTED** [review]
Let's split the image name refactoring out from the custom child prefix part

File: lib/manager/mix/artifacts.ts
Comment (viceice) [reply lib/manager/mix/artifacts.ts:50]: Yes, looks like some missing refactoring here.

**Comment by viceice** [comment]
Blocked by #9132 and #9133

**Comment by viceice** [comment]
No, needs deconflicting now

File: lib/config/definitions.ts
Comment (viceice) [inline lib/config/definitions.ts:259]:
```suggestion
    admin: true,
    default: 'renovate_',
```

Set default, so it's visible in docs.

File: lib/util/exec/docker/index.ts
Comment (viceice) [inline lib/util/exec/docker/index.ts:155]:
```suggestion
```
I think this is tested 🤔

Comment (Chumper) [reply lib/util/exec/docker/index.ts:155]: Nope, unfortunately not :)

File: lib/util/exec/index.ts
Comment (viceice) [inline lib/util/exec/index.ts:99]:
```suggestion
  const { dockerChildPrefix, customEnvVariables } = getAdminConfig();
  const extraEnv = { ...opts.extraEnv, ...customEnvVariables };
```
Simplify

**Review by viceice — COMMENTED** [review]
just a small issue

**Review by viceice — APPROVED** [review]

**Comment by viceice** [comment]
@rarkins needs your approval

**Review by rarkins — APPROVED** [review]

**Pull request merged**
